app/livechat/server/api/v1/agent.js

```diff
@@ -76,3 +77,25 @@ API.v1.addRoute('livechat/agent.next/:token', {
 		}
 	},
 });
+
+API.v1.addRoute('livechat/agents/:agentId/departments', { authRequired: true }, {
```

Comment:
The `app/livechat/server/api/v1/` folder is not the right folder to add Oauth endpoints, in this folder we store only REST endpoints to support the widget.
The right folder is: https://github.com/RocketChat/Rocket.Chat/tree/develop/app/livechat/imports/server/rest

Also, IMO you should rename the endpoint to: 
```suggestion
API.v1.addRoute('livechat/agent/:agentId/departments', { authRequired: true }, {
```